dataplug-spotify/app/org/hatdex/dataplugSpotify/apiInterfaces/SpotifyUserPlaylistsInterface.scala

```diff
@@ -0,0 +1,150 @@
+
+  def buildContinuation(content: JsValue, params: ApiEndpointCall): Option[ApiEndpointCall] = {
+    logger.debug("Building continuation...")
+
+    val nextQueryParams = for {
+      nextLink <- Try((content \ "next").asOpt[String].getOrElse("")) if nextLink.nonEmpty
+      queryParams <- Try(Uri(nextLink).query().toMap) if queryParams.size == 2
+    } yield queryParams
+
+    (nextQueryParams, params.storage.get("offset")) match {
+      case (Success(qp), Some(_)) =>
+        logger.debug(s"Next continuation params: $qp")
+        Some(params.copy(queryParameters = params.queryParameters ++ qp))
+
+      case (Success(qp), None) =>
+        val offsetParameter = (content \ "offset").as[Int].toString
+        logger.debug(s"Next continuation params: $qp, setting offset to $offsetParameter")
+        Some(params.copy(queryParameters = params.queryParameters ++ qp, storageParameters = Some(params.storage +
+          ("offset" -> offsetParameter))))
+
+      case (Failure(e), _) =>
+        logger.debug(s"Next link NOT found. Terminating continuation. $e")
+        None
+    }
+  }
+
+  def buildNextSync(content: JsValue, params: ApiEndpointCall): ApiEndpointCall = {
+    logger.debug(s"Building next sync...")
+
+    params.storage.get("offset").map { offSet => // After date set (there was at least one continuation step)
+      Try((content \ "items").as[JsArray].value) match {
+        case Success(_) => // Did continuation but there was no `nextPage` found, if track array present it's a successful completion
+          val nextStorage = params.storage - "offset"
+          val nextQuery = params.queryParameters - "before" + ("offset" -> offSet)
+          params.copy(queryParameters = nextQuery, storageParameters = Some(nextStorage))
+
+        case Failure(e) => // Cannot extract tracks array value, most likely an error was returned by the API, continue from where finished previously
+          logger.error(s"Provider API request error while performing continuation: $content.The error is: $e")
+          params
+      }
+    }.getOrElse { // After date not set, no continuation steps took place
+      val offsetParameter = (content \ "offset").as[Int].toString
+      val nextQuery = params.queryParameters - "before" + ("offset" -> offsetParameter)
+      params.copy(queryParameters = nextQuery)
+    }
+  }
+
+  private def generateChangedPlaylistIds(content: JsValue, fetchParameters: ApiEndpointCall): Map[String, String] = {
+    val savedSnapshotIdStorage = fetchParameters.storageParameters.getOrElse(Map.empty[String, String])
+    val idsToAdd: Map[String, String] = Map.empty[String, String]
+
+    (content \ "items").as[JsArray].value.map { playlist =>
+      val newSnapshotId = (playlist \ "snapshot_id").as[String]
+      val playlistId = (playlist \ "id").as[String]
+      val maybeSnapshotId = savedSnapshotIdStorage.find(_._1 == playlistId)
+      maybeSnapshotId match {
+        case Some(snapshot) if (snapshot._2 != newSnapshotId) =>
```

Comment:
Looks like you've fallen back to the mutable approach here and it actually doesn't work as expected. `idsToAdd` variable is actually never updated, it just maps JsArray to a Seq of Maps :) Look into using `collect` || `fold`.